libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/ApolloCompiler.kt

```diff
@@ -118,18 +128,28 @@ object ApolloCompiler {
 
     val schema = result.value!!
 
+    val scalarMapping = codegenSchemaOptions.scalarMapping
     checkScalars(schema, scalarMapping)
+
+    val codegenModels = codegenModels(codegenModels = codegenSchemaOptions.codegenModels, codegenSchemaOptions.targetLanguage)
+    @Suppress("NAME_SHADOWING")
+    val packageNameGenerator = packageNameGenerator ?: packageNameGenerator(
+        codegenSchemaOptions.packageName,
+        codegenSchemaOptions.packageNamesFromFilePaths,
+        packageNameRoots
+    )
```

Comment:
This is the prep work for moving `packageNameGenerator` to a `ServiceLoader`. We would still support `packageName` and `packageNamesFromFilePaths` as declarative options present in the JSON files that can be overriden by a compiler plugin if needed.